### PR TITLE
ODC-7583: Add filter to fetch from cluster and tekton results in PipelineRun list pages

### PR DESCRIPTION
**Pull request by lokanandaprabhu**
**Story:**
https://issues.redhat.com/browse/ODC-7583

**Video:**
https://drive.google.com/file/d/1FUmMP_Kf571eClyjrJYHXshqu0qNpJ3W/view?usp=drive_link

**Description:**
1. Added another filter group for the PipelineRun list pages which will have 2 options, one is to fetch live data from cluster and another one is to fetch archived data from tekton results. 

2. If Tekton Results is not installed, then the data source filter group will not be visible

3. By default, cluster data source will be selected

4. Changes are done in below pages
     a) PipelineRun list page
     b) PipelineRun list page tab in Pipeline details page
     c)  PipelineRun list page tab in Repository details page

### CODE REVIEW

File: src/components/utils/pipeline-filter-reducer.ts [hunk 1]
```diff
@@ -146,6 +146,42 @@ export const pipelineRunFilterReducer = (pipelineRun): ComputedStatus => {
   return status || ComputedStatus.Other;
 };
 
+export const pipelineRunDataSourceFilterReducer = (pipelineRun) => {
+  if (
+    pipelineRun?.metadata?.annotations?.['resource.deleted.in.k8s'] &&
+    pipelineRun?.metadata?.annotations?.['resource.loaded.from.tektonResults']
```
[the remark this answers is not in the document]
Comment (lokanandaprabhu) [reply src/components/utils/pipeline-filter-reducer.ts:152]: Updated the PR

File: src/components/pipelineRuns-list/usePipelineRunsFilters.ts [hunk 2]
```diff
@@ -1,37 +1,93 @@
...
+            {
+              id: ListFilterId.Cancelled,
+              title: ListFilterLabels[ListFilterId.Cancelled],
+            },
+            {
+              id: ListFilterId.Other,
+              title: ListFilterLabels[ListFilterId.Other],
+            },
+          ],
+          filter: pipelineRunStatusFilter,
```
[the remark this answers is not in the document]
Comment (lokanandaprabhu) [reply src/components/pipelineRuns-list/usePipelineRunsFilters.ts]: Updated the PR

File: src/components/hooks/useTektonResult.ts [hunk 3]
```diff
@@ -143,30 +141,34 @@ export const useGetPipelineRuns = (
...
-      ? uniqBy([...k8sPlrs, ...resultPlrs], (r) => r.metadata.uid)
-      : [];
-  return [mergedPlrs, k8sPlrsLoaded, k8sPlrsLoadError, getNextPage];
+  return React.useMemo(
+    () => [pipelineRuns, loaded, error, getNextPage],
+    [pipelineRuns, loaded, error, getNextPage],
+  );
 };
 
+export const usePipelineRuns2 = (
```
[the remark this answers is not in the document]
Comment (lokanandaprabhu) [reply src/components/hooks/useTektonResult.ts]: Removed usePipelineRuns2

**Comment by lokanandaprabhu** [comment]
Hi @vikram-raj ,

Updated the PR based on review comments. PTAL.

**Review by vikram-raj — COMMENTED** [review]
/lgtm

**Pull request merged**